### ISSUE TITLE
Warning reported for assignment to property marked [AllowNull]

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#39926]
No warnings should be reported:
```C#
#nullable enable

using System.Diagnostics.CodeAnalysis;

class C<T>
{
    [MaybeNull, AllowNull]
    T F = default;
    [MaybeNull, AllowNull]
    T P { get; } = default; // warning CS8601: Possible null reference assignment

    C([AllowNull]T t)
    {
        F = t;
        P = t; // warning CS8601: Possible null reference assignment
    }
}
```
See also https://github.com/dotnet/roslyn/issues/39922.
